src/models/instance.ts

```diff
@@ -119,40 +120,42 @@ export default class Instance {
         return new Instance({ ...this.data, description });
     }
 
-    public encryptPassword(encryptionKey: string) {
-        const cryptr = new Cryptr(encryptionKey);
-        const encrypted = cryptr.encrypt(this.data.password);
-        return new Instance({ ...this.data, password: encrypted });
+    public async encryptPassword(): Promise<Instance> {
+        const encryptionKey = await Instance.getEncryptionKey();
```

Comment:
It's a good option, I tried adding memoization but couldn't find how to add it with promise based methods

Reply:
Implemented your proposal, as a note if we expand the usage of configuration based properties it might be helpful to create a Class that acts as a configuration file parser with file cache. It's fine for now since we do not store too much info there yet.